internal/envtest/environment.go

```diff
@@ -75,6 +77,52 @@ func init() {
 	utilruntime.Must(admissionv1.AddToScheme(scheme.Scheme))
 }
 
+// RunTestsInput is the input for RunTests.
+type RunTestsInput struct {
+	M                   *testing.M
+	Env                 **Environment
+	ManagerUncachedObjs []client.Object
+	PreManagerStart     func()
+}
+
+// RunTests executes the tests of the given testing.M in a test environment.
+// Note: The environment will be created in this func and should not be created before. This func takes a **Environment
+//       because our tests require access to the *Environment. We use this field to make the created Environment available
+//       to the consumer.
+// Note: Test environment creation can be skipped by setting the environment variable `CAPI_DISABLE_TEST_ENV`. This only
+//       makes sense when executing tests which don't require the test environment, e.g. tests using only the fake client.
+func RunTests(ctx context.Context, input RunTestsInput) int {
```

Comment:
It would be nice if this can be a method of env, so it would read

```
env.RunTests(m)
```

Reply:
I tried this but it get's really complicated. The reason is because this would be used like this:

```go
env := envtest.New()
env.RunTests(m)
```

The problem is that envtest.New() already creates the kind cluster which makes it impossible to skip the envtest creation. I could work around that by adjusting envtest.New that it doesn't really create a kind cluster when `CAPI_DISABLE_TEST_ENV` but this feels counter-intuitive to me.

Reply:
I was assuming to have RunTests as of now; it will call envtest.New() internally if env is nil (and if env test is not disabled)

Reply:
Ah good point, ups :)

Reply:
I think it doesn't work that way because I cannot overwrite the pointer receiver:

```go
func (e *Environment) RunTests(ctx context.Context, input RunTestsInput) int {
	if os.Getenv("CAPI_DISABLE_TEST_ENV") != "" {
		return input.M.Run()
	}

	// Bootstrapping test environment
	if e == nil {
		e = New(input.ManagerUncachedObjs...)
	}
```
Intellij warning: Assignment to the method receiver propagates only to callees not to callers. 

It behaves exactly like that the environment of the caller is still nil.

Or did I just implement it incorrectly?

We can also play around with it synchronously via Zoom if you want, might be more efficient.

Reply:
many thanks for investigating.
I'm not super happing of env being passed back to a method in env test, but let's stick to current solution unless a better idea pops up

Reply:
Yup I also don't like it. The reason for passing the env back via the pointer is mainly because we need the env in all our tests (during m.RunTests()) and I cannot think of a way around it unfortunately